Fix #328. Merge plugin upstream master into 202

sdubov opened the pull request:
* Fix Utils, Plugin modules
* Move AzureActionsListener into extension to support correct services registration for Java and Rider
* Fix plugin.xml - split actions to a separate groups, merged missing actions and clean up idea plugin.xml file (remove actions that were registered in general platformPlugin.xml file)
* Fix Node2Actions map for Java and Rider plugins
* Fix build.gradle - move :idea and general logic into plugin.xml (idea) and platformPlugin.xml accordingly, and move dependent plugin versions into gradle.properties
* Fix package naming and code location for Functions

Fixes #328
Fixes #269
Fixes #281
Fixes #282

t-rufang committed:
Remove deprecated emulator code to prepare for Guava library upgrade

Commit:
1. add spring cloud auth to AuthManager

Commit:
1. adjust unified style of accessing app using appId
2. remove the unnecessary check for null azure spring cloud client

Commit:
1. delete un-finished SpringCloudUtils.java

andxu committed:
Merge branch 'develop' into spring-dev1

andxu committed:
add spring cloud auth to AuthManager (#4178)

* 1. add spring cloud auth to AuthManager
2. fix some checkstyle errors in Node
3. update the node name when it is necessary
4. add AzureSpringCloudMvpModel

* 1. add spring cloud auth to AuthManager

* 1. adjust unified style of accessing app using appId
2. remove the unnecessary check for null azure spring cloud client

* 1. delete un-finished SpringCloudUtils.java

Co-authored-by: Andy Xu(devdiv) <andxu@microsoft>

wezhang committed:
Update IntelliJ project inspection rules

Flanker32 committed:
Fix UI issues in EAP (#4184)

andxu committed:
update azure&checkstyle dependencies (#4185)

* change azure version inside project

* update versions of azure dependencies

* remove wildfly and fix the deprecated api in guava and fix compilation errors caused by version update of azure-tools-common

* revert the removal of wildfly image

* fix the deprecated api in guava

* remove the .repository

Co-authored-by: Andy Xu(devdiv) <andxu@microsoft>

Flanker32 committed:
Show info dialog after create service principal (#4128)

* Show info dialog after create service principal

* Fix wording

Co-authored-by: Andy Xu(devdiv) <andxu@microsoft.com>

Commit:
check in checkstyle rules

wezhang committed:
Enable IntelliJ inspection used import rule

wezhang committed:
Fix coding style issues (#4186)

* Fix codingstyle issues

* Fix TODO checkstyle violations

* Revert Nullable annotation change for checkstyle fixing

andxu committed:
Merge branch 'develop' into andy-rule2

Commit:
The order of factories may change, so need to filter the classname: FunctionRunConfigurationFactory

andxu committed:
Merge pull request #4209 from andxu/andy-functionrun

The order of factories may change, so need to filter the classname

Commit:
Merge remote-tracking branch 'upstream/develop' into andy-rule2

Commit:
Address PR review comments

Commit:
Remove useless HelpMojo.java

Flanker32 committed:
Fix method to get function keys (#4219)

* Fix method to get function keys

* Fix wording

andxu committed:
Merge branch 'develop' into andy-rule2

andxu committed:
Merge pull request #4187 from andxu/andy-rule2

check in checkstyle rules

t-rufang committed:
Routine work: fix coding style (#4212)

* Routine work: fix coding style

* Update based on project coding style requirements

* Fix Wrong lexicographical order issue

* Update based on review comments

Co-authored-by: Zhang Wei <zhwe@microsoft.com>

Flanker32 committed:
Enable Trigger TimerTrigger in Azure Explorer (#4235)

Flanker32 committed:
Set default values for new app service plan (#4240)

* Set default values for new app service plan

* Fix wording

Flanker32 committed:
Update validation in function creation dialog (#4239)

* Change function creation dialog to AzureDialogWrapper for validation

* Fix load subscription multi times

* Remove unused import

* resolve comments

wezhang committed:
Fix coding style issues

t-rufang committed:
Fix NPE caused job submission failure in EAP

t-rufang committed:
Upgrade intelliJ and plugins to the latest version for EAP

Flanker32 committed:
Integrate spring cloud feature (#4261)

* add spring cloud in Azure Explorer

* add dependencies

* add default showConfirmation impl for building eclipse bit

* add default showConfirmation impl for building eclipse bit

* Fix typo 'noDataRefresh'

* avoid to use ConcurrentHashMap in single thread codes

* manage azure spring cloud dependencies.

* address code review comments:
1. Fix field position in DependencyArtifact
2. Avoid so many checkCanceled

* reorganize the code to get latest versions

* Add `open in portal` for spring cloud explorer node (#4210)

* Add spring cloud deploy features:
1. add deploy configuration and setting editor
2. add deploy runner
3. refactor some common constants

* fix checkstyle errors.

* fix checkstyle error

* 1. use the right string utility
2. add plugin.xml

* Fix issues in spring deployment (#4217)

* Pending deployment done and show url

* Use copy constructor when generate spring run state and show error form when deploy failed

* Add null check and fix checkstyle issues

* 1. Use effective pom to get proper jar name
2. Replace the confusing name moduleName(actually targetName)
3. Remove getTargetPath getArtifactPath getModuleName in Spring Configuration, there are all useless
4. Add projectName in Spring Configuration

* 1. Rename IdHelper to SpringIdHelper
2. Fix the status synchronization with app -> parent(cluster)

* revert changes back

* Add log streaming for spring cloud (#4226)

* add updateAppProperties

* Add environment table component (#4236)

* Fix saved env variables can't be modified (#4241)

* Spring Cloud property view (#4238)

* Spring Cloud property view

* Add env component in property view (#4250)

* Fix the issue: missing <Empty> options after switching clusters.

* Mark border when the value is dirty

* avoid to use ConcurrentHashMap since null cannot be inserted as value

* Freeze UI and clear mark during update

* disable java version combo editing

* Add env component in property view

Co-authored-by: Andy Xu(devdiv) <andxu@microsoft>

* Add support for function streaming log (#4246)

* Fix prompt messages in spring cloud log streaming (#4258)

* Fix set text when process handler is terminated (#4256)

* Fix set text when process handler is terminated

* Add new icons for spring cloud dependency (#4259)

* Add new icons for spring dependency

* Add validation for spring cloud deploy configuration (#4255)

* Add validation for spring cloud deploy configuration

* Resolve comments

* Move lightly validation ahead

* Fix not available label (#4254)

* 1. Fix the println error when process is exited
2. Use the disable for bad hint message: ' - '
3. add preview to spring cloud node
4. use a better singleton way for DefaultAzureResourceTracker
5. fix the Maps.difference null issue

* Add telemetry for spring cloud actions (#4260)

* Add telemetry for spring cloud actions

* Fix env table trigger issue

Co-authored-by: Andy Xu(devdiv) <andxu@microsoft.com>

* Fix merge conflict

Co-authored-by: Andy Xu(devdiv) <andxu@microsoft>
Co-authored-by: Andy Xu(devdiv) <andxu@microsoft.com>

Flanker32 committed:
Fix map compare methods (#4267)

Flanker32 committed:
Prepare release IntelliJ Toolkit 3.35.0 (#4266)

* Bump version for release

* Add change log for intellij toolkit 3.35.0

* Add change log for function log streaming

t-rufang committed:
Fix IntelliJ bundle build cannot open project issue (#4265)

* Fix IntelliJ bundle build cannot open project issue

* Fix build break

* Fix illegal file name parsed from JBR URL issue

Flanker32 committed:
Fix UI thread issue (#4269)

Flanker32 committed:
Fix configuration border cases (#4273)

Flanker32 committed:
Support trigger function event hub trigger (#4274)

* Support trigger function event hub trigger

* Remove unused codes

* Move get master key to function node and distinguish admin and function auth level

t-rufang committed:
Tag log source when print logs to console view for spark jobs

Flanker32 committed:
Redesign spring cloud property view (#4291)

* Fix UI issues

* Surround instances table with jscrollpane

* Remove unused size settings and better naming

Flanker32 committed:
Update app properties after active deployment (#4295)

* Update app properties after active deployment

* Better naming

Flanker32 committed:
Merge branch 'master' into develop

t-rufang committed:
Fix coding style (#4292)

* Fix coding style

* Fix check style build error

Flanker32 committed:
Merge branch 'develop' into develop.next

Flanker32 committed:
Merge conflicts

andxu committed:
enable SP for spring cloud (#4303)

2. Fix issue https://github.com/microsoft/azure-tools-for-java/issues/4230

Co-authored-by: Andy Xu(devdiv) <andxu@microsoft>

wezhang committed:
Remove IntelliJ inspection ERROR for changed file

Also remove methods parameter `final` inspection warning report

andxu committed:
Don't generate version when the version is in dependencyManagement se… (#4305)

* Don't generate version when the version is in dependencyManagement section.

andxu committed:
Remove complex loading for effective since it is already implemented in non-cancelable way. (#4308)

Co-authored-by: Andy Xu(devdiv) <andxu@microsoft>

Flanker32 committed:
Fix icon deprecated issue in IntelliJ 2020.1 (#4311)

t-rufang committed:
Fix bug introduced by coding style PR

Flanker32 committed:
Enable log streaming for app service and refactor codes (#4315)

andxu committed:
Try to use version in <dependencyManagement> rather than latest m2 version. (#4317)

Co-authored-by: Andy Xu(devdiv) <andxu@microsoft>

t-rufang committed:
Fix Eclipse build break issue

Flanker32 committed:
Add right click pop up in spring cloud property view for log streaming (#4321)

* Add right click pop up in spring cloud property view for log streaming

* Fix checkstyle and add telemetry

* Set log streaming menu enable status

* Fix logic to hide popup menu when right click blank area

* Control jtable popup in mouse listener

Flanker32 committed:
Open portal Live Metrics Stream for linux function app  (#4320)

* Add log streaming support for linux function app

* Fix checkstyle

* Fix wording

* Fix exception handle

* Fix checkstyle

Flanker32 committed:
Add status to spring console view (#4322)

* Add status to spring console view

* Fix thread logic

* Fix logic handling error while getting log streaming content

* Fix start log logic

t-rufang committed:
Update based on review comments

Flanker32 committed:
Trace run configuration parameters (#4324)

Flanker32 committed:
Fix spring run configuration issues (#4323)

* Fix spring run configuration issues

* Update spring configuration once previous parameters become invalid

Co-authored-by: Andy Xu(devdiv) <andxu@microsoft.com>

andxu committed:
Update the error message when the version is implied from dependencyManagement configuration (#4326)

Co-authored-by: Andy Xu(devdiv) <andxu@microsoft>

Flanker32 committed:
Add title for Eclipse dialog (#4325)

* Add title for Eclipse dialog

* Fix checkstyle issues

* Extract string into constants

t-rufang committed:
Fix coding style caused build break

t-rufang committed:
Update based on review comments

t-rufang committed:
Update based on review comments

pablomh committed:
Fix typo (#4329)

occured -> occurred.

Flanker32 committed:
Add artifact validation when deploy spring cloud app (#4330)

* Add artifact validation when deploy spring cloud app

* Fix wording

* Resolve comments

* Using isEmpty instead of size > 0

Flanker32 committed:
Add change log for intellij toolkits 3.36.0 (#4332)

Flanker32 committed:
Bump version for release (#4331)

Co-authored-by: Andy Xu(devdiv) <andxu@microsoft.com>

t-rufang committed:
Merge branch 'develop' into develop.next

t-rufang committed:
Fix resolve merge conflict mistake

t-rufang committed:
Fix check style error caused build break

Flanker32 committed:
Merge pull request #4337 from t-rufang/develop.next

Merge develop into develop.next

Flanker32 committed:
Merge branch 'release.next' into develop.next

Flanker32 committed:
Update plugin description

Flanker32 committed:
Fix wording

Flanker32 committed:
Merge pull request #4339 from Flanker32/hanxiao/description

Update plugin description

Flanker32 committed:
Use IntelliJ Messages API to replace JOptionPane

t-rufang committed:
Fix uploading jar with interactive session caused job submission failure issue

Flanker32 committed:
Fix icon and checkstyle

Flanker32 committed:
Fix checkstyle issues

Flanker32 committed:
Rename `Deploy to Azure` action to `Deploy to Azure Web Apps`

Flanker32 committed:
Merge pull request #4348 from Flanker32/hanxiao/rename

Rename `Deploy to Azure` action to `Deploy to Azure Web Apps`

Flanker32 committed:
Merge pull request #4343 from Flanker32/hanxiao/removeOptionPane

Use IntelliJ Messages API to replace JOptionPane

Flanker32 committed:
Update dependencies in Utils

Flanker32 committed:
Fix show messages dialog will throw exception in non-dispatch thread

Flanker32 committed:
Remove unused imports

Flanker32 committed:
Fix code style and better naming

Flanker32 committed:
Remove duplicated codes

Flanker32 committed:
Merge pull request #4354 from Flanker32/hanxiao/dispatch

Fix show messages dialog will throw exception in non-dispatch thread

Flanker32 committed:
Merge branch 'release' into release.next

Flanker32 committed:
Update gradle dependencies

Flanker32 committed:
Fix NPE when show confirm dialog

Flanker32 committed:
Merge pull request #4359 from Flanker32/hanxiao/fixNPE

Fix NPE when show confirm dialog

Flanker32 committed:
Merge branch 'release' into release.next

Flanker32 committed:
Add dependency validation plugin and update gradle to 5.0

t-rufang committed:
Sort SoC serverless job list by state and submitTime (#4363)

* Sort SoC serverless job list by state and submitTime

* Update based on review comments

Flanker32 committed:
Add whats new page in toolkits

Flanker32 committed:
Merge branch 'develop' into hanxiao/whatsnew

Flanker32 committed:
Add license header

Flanker32 committed:
Merge branch 'hanxiao/whatsnew' of github.com:Flanker32/azure-tools-for-java into hanxiao/whatsnew

Flanker32 committed:
Remove unused constants

Flanker32 committed:
Filter unsupported region while list regions

Flanker32 committed:
Add empty subs check

Flanker32 committed:
Add validation for webapp and add validation chache

andxu committed:
Merge branch 'develop' into hanxiao/update

Flanker32 committed:
Merge pull request #4358 from Flanker32/hanxiao/update

Update IntelliJ Dependencies

Flanker32 committed:
Merge branch 'develop' into hanxiao/addRegionValidation

Flanker32 committed:
Fix code style

Flanker32 committed:
Fix NPE

Flanker32 committed:
Merge pull request #4367 from Flanker32/hanxiao/addRegionValidation

Use "Microsoft.Web/geoRegions" API filter unsupported region

Flanker32 committed:
Merge branch 'develop' into hanxiao/whatsnew

Flanker32 committed:
Refactor throw exception method

Flanker32 committed:
Invoke ui operations with invoke and wait

Flanker32 committed:
Set whats new file read only

Flanker32 committed:
Fix NPE while getting action status

Flanker32 committed:
Resolve comments

Flanker32 committed:
Fix checkstyle

Flanker32 committed:
Merge pull request #4368 from Flanker32/hanxiao/updateValidation

Add validation for webapp and add validation cache

Flanker32 committed:
Merge branch 'develop' into hanxiao/whatsnewmark

Flanker32 committed:
Merge branch 'develop' into hanxiao/fixActionNPE

Flanker32 committed:
Merge branch 'develop' into hanxiao/fixUIIssue

Flanker32 committed:
Fix vm creation will prompt error message twice

t-rufang committed:
Set SparkSubmission folder and artifact file permission to 755 for Gen2 deploy

t-rufang committed:
Refactor SharedKeyObservable

Flanker32 committed:
Fix NPE in eclipse explorer

Flanker32 committed:
Fix docker client dependency

Flanker32 committed:
Merge branch 'develop' into hanxiao/fixEclipse1

Flanker32 committed:
Fix code style

Flanker32 committed:
Merge branch 'hanxiao/fixEclipse1' of github.com:Flanker32/azure-tools-for-java into hanxiao/fixEclipse1

Flanker32 committed:
Refactor validate method

Flanker32 committed:
Fix wording

Flanker32 committed:
Merge pull request #4376 from Flanker32/hanxiao/fixEclipse1

Fix Eclipse issues found in regression test

andxu committed:
Merge branch 'develop' into hanxiao/fixUIIssue

Flanker32 committed:
Bump version for release

Flanker32 committed:
Merge branch 'develop' into hanxiao/whatsnewmark

Flanker32 committed:
Merge branch 'develop' into hanxiao/fixActionNPE

Flanker32 committed:
Merge pull request #4371 from Flanker32/hanxiao/fixUIIssue

Fix function deployment panel initialization issue

Flanker32 committed:
Merge branch 'develop' into hanxiao/fixActionNPE

Flanker32 committed:
Merge branch 'develop' into hanxiao/whatsnewmark

Flanker32 committed:
Merge branch 'develop' into hanxiao/bump_version

Flanker32 committed:
Merge pull request #4373 from Flanker32/hanxiao/fixActionNPE

Fix NPE while getting action status

Flanker32 committed:
Merge branch 'develop' into hanxiao/bump_version

Flanker32 committed:
Merge branch 'develop' into hanxiao/whatsnewmark

Flanker32 committed:
Better naming

Flanker32 committed:
Merge branch 'hanxiao/whatsnewmark' of github.com:Flanker32/azure-tools-for-java into hanxiao/whatsnewmark

t-rufang committed:
Fix upsupported header in request with OAuth as authorization method when create file and fodler in Gen2

t-rufang committed:
Fix term Spark usage in toolkit UI

Flanker32 committed:
Merge pull request #4377 from Flanker32/hanxiao/bump_version

Bump version for release

Flanker32 committed:
Merge branch 'develop' into hanxiao/whatsnewmark

t-rufang committed:
Update based on review comments

Flanker32 committed:
Resolve comments

Flanker32 committed:
Merge branch 'hanxiao/whatsnewmark' of github.com:Flanker32/azure-tools-for-java into hanxiao/whatsnewmark

t-rufang committed:
Replace null list with empty list

Flanker32 committed:
Merge pull request #4372 from Flanker32/hanxiao/whatsnewmark

Add whats new page in toolkits(Markdown)

Flanker32 committed:
Set `App Settings` label vertical align to top in function deployment panel, keep consistent with run panel

Flanker32 committed:
Merge branch 'develop' into hanxiao/fixUI

Flanker32 committed:
Add spring cluster validation cache

Flanker32 committed:
Merge pull request #4378 from Flanker32/hanxiao/fixUI

Set `App Settings` label vertical align to top in deployment panel

Flanker32 committed:
Fix used storage count may be null

andxu committed:
Merge branch 'develop' into hanxiao/fixCache

Flanker32 committed:
Fix checkstyle issue

Flanker32 committed:
Merge branch 'hanxiao/fixCache' of github.com:Flanker32/azure-tools-for-java into hanxiao/fixCache

Flanker32 committed:
Update change log and bump version

Flanker32 committed:
Add toc for new features

Flanker32 committed:
Merge pull request #4380 from Flanker32/hanxiao/fixCache

Fix Spring Cloud UI Issues

Flanker32 committed:
Merge branch 'develop' into hanxiao/releasenotes

Flanker32 committed:
Merge pull request #4381 from Flanker32/hanxiao/releasenotes

Update change log and bump version

Flanker32 committed:
Merge branch 'develop' into develop.next

Flanker32 committed:
Fix whats new document issue

Flanker32 committed:
Merge pull request #4383 from Flanker32/hanxiao/whatsnewdoc

Fix whats new document issue

Flanker32 committed:
Merge branch 'develop' into develop.next

andxu committed:
Add some maven args to boost build perf (#4387)

Co-authored-by: Andy Xu(devdiv) <andxu@microsoft>

Flanker32 committed:
Bump Azure version

andxu committed:
Merge branch 'develop' into hanxiao/fixAzureBump

Flanker32 committed:
Fis whats new shown in 2020.1

Flanker32 committed:
Fix pop up dialog layer issue

t-rufang committed:
Fix term Spark usage in plugin description

Flanker32 committed:
Fix Eclipse dependency issue

Flanker32 committed:
Merge pull request #4397 from Flanker32/hanxiao/fixEclipseDependency

Fix Eclipse dependency issue

Flanker32 committed:
Merge branch 'develop' into hanxiao/fixAzureBump

Flanker32 committed:
Merge branch 'develop' into hanxiao/fixWhatsNew2020

Flanker32 committed:
Merge pull request #4391 from Flanker32/hanxiao/fixAzureBump

Fix bump Azure library version issue

Flanker32 committed:
Merge branch 'develop' into hanxiao/fixWhatsNew2020

Flanker32 committed:
Merge pull request #4394 from Flanker32/hanxiao/fixWhatsNew2020

Fix UI issues for IntelliJ Toolkit

andxu committed:
Refresh Azure Spring Cloud service model using RxJava3 (#4386)

* Refresh Azure Spring Cloud service model using RxJava3

Flanker32 committed:
Merge branch 'develop' into develop.next

Flanker32 committed:
Fix release notes markdown title level

Flanker32 committed:
Merge pull request #4408 from Flanker32/hanxiao/fixReleaseNotes

Fix release notes markdown title level

Flanker32 committed:
Merge branch 'develop' into develop.next

t-rufang committed:
Fix no response after click submit spark application option issue

Flanker32 committed:
Fix can not switch to new create app service issue

Flanker32 committed:
Fix run docker issue

t-rufang committed:
Retain IntelliJ version to 2020.1 to avoid CI build failure

Flanker32 committed:
Resolve comments

Flanker32 committed:
Add missing codes

Flanker32 committed:
Merge pull request #4420 from Flanker32/hanxiao/fixIntelliJ

Fix IntelliJ regression issues

Flanker32 committed:
Merge branch 'develop' into develop.next

wezhang committed:
Fix Spark mockfs jar scheme issue

andxu committed:
Fix ConcurrentModificationException (#4426)

Fix issue https://github.com/microsoft/azure-tools-for-java/issues/4422

Flanker32 committed:
Add AI related methods with azure SDK (#4428)

* Add AI related methods with azure SDK

Flanker32 committed:
Add bot to close no response need more info issus (#4429)

Flanker32 committed:
Deprecated old private application insights sdk (#4430)

* Deprecated old private application insights sdk

* Fix checkstyle and remove unused comments

* Remove unused constants

wezhang committed:
Fix Spark local console regression after Scala plugin upgraded to org.intellij.scala:2020.1.35

wezhang committed:
Upgrade dev scala plugin version to 2020.1.36

wezhang committed:
Upgrade pipeline Scala plugin version to 2020.1.36

Commit:
Fix new config in job config table doesn not take effect after click cancle button issue

Flanker32 committed:
Fix Open what's new page modality state issue (#4438)

Flanker32 committed:
Support creating application insights while creating function app (#4434)

Flanker32 committed:
Update change log and bump version (#4447)

* Bump version for IntelliJ and Utils

* Update release notes

Flanker32 committed:
Deprecate docker host (#4446)

t-rufang committed:
Add Azure synapse feature in what's new page (#4444)

* Add Apache branding before word Spark in whats new file

* Add Azure synapse feature in what's new page

* Update based on review comments

* Wording update based on review comments

* Add change log to whats new file

* Fix typo

* Fix gif size too big caused rendering issue

Co-authored-by: Rui Fang <ruifang@Venus.solar.lab>

Flanker32 committed:
Merge branch 'develop' into develop.next

Flanker32 committed:
Fix issues in IntelliJ End game (#4455)

* Fix always creating application insights issue

* Fix spring cloud properties refresh issue

Flanker32 committed:
Merge branch 'release' into release.next

andxu committed:
Fix bug 4389: check-box UI issue when creating function project (#4460)

Co-authored-by: Andy Xu(devdiv) <andxu@microsoft>

andxu committed:
Fix bug 4307: Selecting the wrong module automatically when adding gr… (#4461)

* Fix bug 4307: Selecting the wrong module automatically when adding gradle project.


Co-authored-by: Andy Xu(devdiv) <andxu@microsoft>

andxu committed:
Remove useless code to fix issue #4016: Weak Cryptographic Hash (#4464)

* Remove useless code to fix issue #4016: Weak Cryptographic Hash


Co-authored-by: Andy Xu(devdiv) <andxu@microsoft>

Flanker32 committed:
Fix set text process terminated handler (#4465)

* Fix set text process terminated handler, do nothing in onSuccess/OnFail

* Better naming and null handling

t-rufang committed:
Show Apache Spark on Cosmos node by default (#4467)

Flanker32 committed:
Fix AuthException when account contains no subscription (#4458)

andxu committed:
Add common modules back which should be supported in Azure Functions, Fix #4307 (#4466)

Co-authored-by: Andy Xu(devdiv) <andxu@microsoft>
Co-authored-by: Hanxiao Liu <hanli@microsoft.com>

Flanker32 committed:
Merge branch 'release' into develop

Flanker32 committed:
Bump dependencies (#4470)

Flanker32 committed:
Fix init multi times issue (#4471)

* Fix init multi times issue

* Fix checkstyle issue

Flanker32 committed:
Support Azure CLI credential for Toolkit auth (#4468)

* Support Azure CLI credential for Toolkit auth

* Fix azure cli auth can't save subscription selection

* Resolve comments

* Fix wording and resolve comments

* Show Azure Cli but disable it in sign in window if not login

t-rufang committed:
Fix hdi cluster false recognized as reader role cluster when signin with azure cli

Flanker32 committed:
Fix AuthMethod clear issue (#4473)

* Fix AuthMethod clear issue

* Fix Null Handling

Commit:
Fix Spark history server link break issue

Flanker32 committed:
Fix NPE while parse function annotations (#4477)

Flanker32 committed:
Fix NPE while refresh Azure (#4479)

Commit:
Update synapse feature in whats new page

Flanker32 committed:
Bump version for release (#4483)

Flanker32 committed:
Support Java 11 Azure Functions (#4482)

* Add support for java 11

* Update Function Creation dialog UI design

* Fix prompt message

* Remove unused methods

* Remove outdated comments

Flanker32 committed:
Remove docker host and fix azure library (#4484)

Flanker32 committed:
Add release notes for intellij toolkit v3.39.0 (#4485)

* Add release notes for new release

* Fix wording issues

Commit:
Fix typo

Flanker32 committed:
Remove snapshot (#4486)

Flanker32 committed:
Merge branch 'develop' into develop.next

Flanker32 committed:
Fix wording in login window (#4495)

Flanker32 committed:
Bump dependencies version (#4500)

andxu committed:
Merge branch 'develop' into develop.next

t-rufang committed:
Upgrade develop branch to IntelliJ 2020.1

Flanker32 committed:
Fix typo (#4506)

Co-authored-by: Andy Xu(devdiv) <andxu@microsoft.com>

Flanker32 committed:
Fix azure dependencies version (#4512)

Flanker32 committed:
Bump version and update change log for release (#4516)

* Bump version for release

* Update change log

Flanker32 committed:
List anonymous HTTP triggers after function deployment (#4508)

* List annoymous http triggers after function deployment

* Provide function access keys document after non-anonymous triggers deployed

* Adopt stable tools common library

sdubov committed:
Merge branch 'master' of github.com:microsoft/azure-tools-for-java into 202

 Conflicts:
	BuildDevint
	BuildProduct.sh
	PluginsAndFeatures/azure-toolkit-for-intellij/.gitignore
	PluginsAndFeatures/azure-toolkit-for-intellij/build.gradle
	PluginsAndFeatures/azure-toolkit-for-intellij/gradle.properties
	PluginsAndFeatures/azure-toolkit-for-intellij/gradle/wrapper/gradle-wrapper.properties
	PluginsAndFeatures/azure-toolkit-for-intellij/idea/resources/META-INF/plugin.xml
	PluginsAndFeatures/azure-toolkit-for-intellij/idea/src/com/microsoft/azure/hdinsight/jobs/JobViewApplicationComponent.java
	PluginsAndFeatures/azure-toolkit-for-intellij/idea/src/com/microsoft/azure/hdinsight/projects/HDInsightActionsComponent.java
	PluginsAndFeatures/azure-toolkit-for-intellij/idea/src/com/microsoft/azure/hdinsight/serverexplore/action/AddNewEmulatorAction.java
	PluginsAndFeatures/azure-toolkit-for-intellij/idea/src/com/microsoft/azure/hdinsight/serverexplore/ui/AddNewEmulatorForm.form
	PluginsAndFeatures/azure-toolkit-for-intellij/idea/src/com/microsoft/azure/hdinsight/serverexplore/ui/AddNewEmulatorForm.java
	PluginsAndFeatures/azure-toolkit-for-intellij/idea/src/com/microsoft/azure/hdinsight/spark/console/SparkScalaLivyConsole.kt
	PluginsAndFeatures/azure-toolkit-for-intellij/idea/src/com/microsoft/azure/hdinsight/spark/run/SparkBatchJobDebugExecutor.java
	PluginsAndFeatures/azure-toolkit-for-intellij/idea/src/com/microsoft/azure/hdinsight/spark/run/SparkBatchJobDebuggerRunner.java
	PluginsAndFeatures/azure-toolkit-for-intellij/idea/src/com/microsoft/azure/hdinsight/spark/run/action/SparkJobDebugAction.kt
	PluginsAndFeatures/azure-toolkit-for-intellij/idea/src/com/microsoft/azure/hdinsight/spark/run/action/SparkJobRunAction.kt
	PluginsAndFeatures/azure-toolkit-for-intellij/idea/src/com/microsoft/intellij/runner/webapp/webappconfig/WebAppRunState.java
	PluginsAndFeatures/azure-toolkit-for-intellij/idea/src/com/microsoft/intellij/wizard/functions/facet/FunctionsFacet.java
	PluginsAndFeatures/azure-toolkit-for-intellij/idea/src/com/microsoft/intellij/wizard/functions/facet/FunctionsFacetConfiguration.java
	PluginsAndFeatures/azure-toolkit-for-intellij/idea/src/com/microsoft/intellij/wizard/functions/facet/FunctionsFacetConfigurationImpl.java
	PluginsAndFeatures/azure-toolkit-for-intellij/idea/src/com/microsoft/intellij/wizard/functions/facet/FunctionsFacetType.java
	PluginsAndFeatures/azure-toolkit-for-intellij/idea/src/com/microsoft/intellij/wizard/functions/module/FunctionTriggerChooserStep.java
	PluginsAndFeatures/azure-toolkit-for-intellij/idea/src/com/microsoft/intellij/wizard/functions/module/FunctionsModuleBuilder.java
	PluginsAndFeatures/azure-toolkit-for-intellij/idea/src/com/microsoft/intellij/wizard/functions/module/FunctionsModuleInfoStep.java
	PluginsAndFeatures/azure-toolkit-for-intellij/idea/test/java/com/microsoft/intellij/tooling/IntegrationTestBase.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/jobs/JobViewApplicationComponent.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/jobs/JobViewApplicationListener.kt
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/projects/HDInsightActionsComponent.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azuretools/ijidea/actions/AzureSignInAction.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azuretools/ijidea/actions/SelectSubscriptionsAction.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azuretools/ijidea/ui/LoginWindow.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/AzureActionsListener.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/AzureConfigurable.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/AzurePlugin.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/actions/AzureDockerHostDeployAction.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/actions/AzureExplorerOpenAction.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/common/CommonConst.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/docker/utils/AzureDockerUIResources.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/docker/wizards/createhost/AzureNewDockerWizardDialog.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/docker/wizards/publish/forms/AzureSelectDockerHostStep.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/feedback/Reportable.kt
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/forms/CreateArmStorageAccountForm.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/forms/TableEntityForm.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/forms/TablesQueryDesigner.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/helpers/IDEHelperImpl.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/helpers/UIHelperImpl.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/helpers/rediscache/RedisCacheExplorer.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/helpers/storage/BlobExplorerFileEditor.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/helpers/storage/TableFileEditor.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/helpers/webapp/WebAppBasePropertyView.form
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/AzureRunConfigurationBase.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/AzureRunProfileState.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/springcloud/deploy/SpringCloudDeploymentRunner.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/serviceexplorer/NodeActionsMap.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/serviceexplorer/azure/ManageSubscriptionsAction.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/serviceexplorer/azure/SignInOutAction.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/serviceexplorer/azure/docker/DeleteDockerHostAction.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/serviceexplorer/azure/rediscache/CreateRedisCacheAction.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/serviceexplorer/azure/storagearm/CreateStorageAccountAction.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/serviceexplorer/azure/vmarm/CreateVMAction.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/ui/AzurePanel.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/ui/components/AzureActionListenerWrapper.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/ui/components/AzureDialogWrapper.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/ui/components/AzureWizardStep.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/util/MethodUtils.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/wizards/createarmvm/SettingsStep.java
	PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/wizards/createarmvm/SubscriptionStep.java
	Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/helpers/UIHelper.java
	Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/Node.java
	Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/RefreshableNode.java
	Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/AzureModule.java
	Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/docker/DockerContainerNode.java
	Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/docker/DockerHostModule.java
	Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/docker/DockerHostNode.java
	Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/docker/DockerImageNode.java
	Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/rediscache/RedisCacheModule.java
	Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/rediscache/RedisCacheNode.java
	Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/storage/ContainerNode.java
	Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/storage/QueueNode.java
	Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/storage/StorageModule.java
	Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/storage/StorageNode.java
	Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/storage/TableNode.java
	Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/storage/asm/StorageNode.java
	Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/vmarm/VMArmModule.java
	Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/vmarm/VMNode.java
	Utils/azure-explorer-common/test/java/com/microsoft/tooling/IntegrationTestBase.java
	Utils/azuretools-core/src/com/microsoft/azuretools/azurecommons/tasks/CancellableTask.java
	Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/ui/base/MvpView.java
	Utils/azuretools-core/src/com/microsoft/azuretools/ui/embeddedbrowser/JxBrowserException.java
	Utils/azuretools-core/src/com/microsoft/azuretools/ui/embeddedbrowser/JxBrowserUtil.java
	Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/common/ClusterManagerEx.java
	Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/common/CommonConst.java
	Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/serverexplore/hdinsightnode/StorageAccountFolderNode.java
	Utils/pom.xml

sdubov committed:
Fix #328. Fix issues and build scripts after merging the upstream master branch

- Fix Utils, Plugin modules
- Move AzureActionsListener into extension to support correct services registration for Java and Rider
- Fix plugin.xml - split actions to a separate groups, merged missing actions and clean up idea plugin.xml file (remove actions that were registered in general platformPlugin.xml file)
- Fix Node2Actions map for Java and Rider plugins
- Fix build.gradle - move :idea and general logic into plugin.xml (idea) and platformPlugin.xml accordingly, and move dependent plugin versions into gradle.properties
- Fix package naming and code location for Functions
- Split AzureModule initialization logic between Java and Rider
- Remove applicationinsights JAR dependency from build scripts since it was removed from Gradle dependency in plugin
- Revert Docker module (need to decide if we will show it in Rider)
- Remove "Deprecated" label from Storage Accounts (need to decide if we should show it)
- Revert changes in BlobExplorerFileEditor class to restore images for blob container
- Revert changes in RedisCacheExplorer

sdubov commented:
Couple notes:
1. Storage Account node is failing at runtime with getting a dependency from `azure-storage`:
```
java.lang.NoClassDefFoundError: Could not initialize class com.microsoft.azure.storage.core.Utility
```
2. Azure Streaming Logs tool window is added. It miss the icon - ([RIDER-48520](https://youtrack.jetbrains.com/issue/RIDER-48520)), but I believe we can include it in a release as well.

maartenba commented:
LGTM apart from those two issues

maartenba reviewed (changes requested):
Storage node expansion seems a blocker

maartenba commented on .github/no-response.yml:
```diff
@@ -0,0 +1,4 @@
+daysUntilClose: 14
```
This will auto-close issues. Is this something we want to do?

sdubov replied:
Definitely not. Thank you for the catch.

sdubov commented:
> Storage node expansion seems a blocker

+1

maartenba committed:
Cloud Shell does not work when using Azure CLI for authentication (version 2020.2+) (#341)

Related to #340

maartenba committed:
Delete no-response.yml

sdubov committed:
Fix NoClassDefFoundError exception on Storage Account node open in Azure Explorer

- Move 'dom4j' and 'jaxen' libraries dependency to idea since it is used for Java-related code. These libs intoduced an error on opening StorageAccount node in Azure Explorer
